pandas/io/gbq.py

```diff
@@ -148,122 +192,201 @@ def run_query(self, query):
             'configuration': {
                 'query': {
                     'query': query
-                    #'allowLargeResults', 'createDisposition', 'preserveNulls', destinationTable, useQueryCache
+                    # 'allowLargeResults', 'createDisposition', 'preserveNulls', destinationTable, useQueryCache
                 }
             }
         }
 
         try:
-            query_reply = job_collection.insert(projectId=self.project_id,
-                                           body=job_data).execute()
-            status = query_reply['status']
+            query_reply = job_collection.insert(projectId=self.project_id, body=job_data).execute()
         except AccessTokenRefreshError:
-            raise AccessDeniedException("The credentials have been revoked or expired, please re-run"
-                "the application to re-authorize")
+            raise AccessDenied("The credentials have been revoked or expired, please re-run the application "
+                                        "to re-authorize")
         except HttpError as ex:
-            status = json.loads(ex.content)['error']
-
-
-        errors = status.get('errors', None)
-
-        if errors:
-            reasons = [error['reason'] for error in errors]
-            if 'accessDenied' in reasons:
-                raise AccessDeniedException
-            if 'invalidQuery' in reasons:
-                raise InvalidQueryException
-            if 'notFound' in reasons:
-                raise NotFoundException
-            if 'termsOfServiceNotAccepted' in reasons:
-                raise TermsOfServiceNotAcceptedException
-            else:
-                raise UnknownGBQException(errors)
+            self.process_http_error(ex)
 
         job_reference = query_reply['jobReference']
 
-        while(not query_reply.get('jobComplete', False)):
-            print('Job not yet complete...')
-            query_reply = job_collection.getQueryResults(
-                            projectId=job_reference['projectId'],
-                            jobId=job_reference['jobId']).execute()
+        while not query_reply.get('jobComplete', False):
+            if verbose:
+                print('Waiting for job to complete...')
```

Comment:
much futher enhancement (future PR), ``verbose`` could be a callable, which you would execute with the message (rather than actual print statements)